test/rebar3_ex_doc_SUITE.erl

```diff
@@ -351,11 +352,27 @@ get_app_metadata(Name, Vsn) ->
         {links, []}
     ]}.
 
+-if(?OTP_RELEASE >= 27).
```

Comment:
Both edoc and doc attributes should work in 27, so maybe you want to test with both?

Reply:
Yeah, we can do that. It's just that when I did `@doc`, as previously, it didn't generate the same output.

Module under test was

```erlang
        "%%%-------------------------------------------------------------------\n"
        "%% @doc `~s' - a module\n"
        "%% @end\n"
        "%%%-------------------------------------------------------------------\n"
        "-module('~s').\n"
        "-export([foo/0]).\n"
        "%%% @doc\n"
        "%%% foo/0 does nothing\n"
        "%%% @end\n"
        "-spec foo() -> ok.\n"
        "foo() -> ok.\n",
        [Name, Name]
```

and there was no generation for `foo/0 does nothing`, unless that one's supposed to be pure EDoc, and not `rebar3_ex_doc`.

Reply:
You need to pass `+nodocs` when compiling the file, otherwise it will create empty doc entries in the beam file and ex_doc will use those instead of the .chunk file that edoc creates.

```erlang
        "%%%-------------------------------------------------------------------\n"
        "%% @doc `~s' - a module\n"
        "%% @end\n"
        "%%%-------------------------------------------------------------------\n"
        "-module('~s').\n"
        "-compile(nodocs).\n"
        "-export([foo/0]).\n"
        "%%% @doc\n"
        "%%% foo/0 does nothing\n"
        "%%% @end\n"
        "-spec foo() -> ok.\n"
        "foo() -> ok.\n",
```

Come to think of it, that is probably something we could and should warn for in edoc_layout_chunks....

Reply:
Actually, I will fix it so that if there are no doc attributes in the code at all, it will be the same as passing `nodocs`, unless `warn_missing_docs` is passed. So the testcase should pass in rc2.

Reply:
With https://github.com/erlang/otp/pull/8192 the test should pass using both edoc and EEP-59 docs.

Reply:
Thanks for this, @garazdawi. I'll check it (your branch) and keep this open, after which maybe we can simply rebase (once your pr gets merged - come to think of it our CI should be targeting the main branch, not a specific tag).